### PR TITLE
Reward Modeling Finetuning Example

### PR DESCRIPTION
**Pull request by Uxito-Ada**
## Description

This is an example of TRL reward modeling, a kind of RLHF, where uses dataset Anthropic/hh-rlhf that on our whitelist.

### 1. Why the change?

Enable RM on Intel GPU with IPEX LLM.

### 2. User API changes

no

### 3. Summary of the change 

Reward Modeling Finetuning Example

### 4. How to test?
- [ ] N/A
- [ ] Unit test
- [x] Application test
- [x] Document test
- [ ] ...

### CODE REVIEW

File: python/llm/example/GPU/LLM-Finetuning/RewardModeling/reward_modeling_finetuning.py [hunk 1]
```diff
@@ -0,0 +1,100 @@
+#
+# Copyright 2016 The BigDL Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
```
Comment (qiyuangong) [inline python/llm/example/GPU/LLM-Finetuning/RewardModeling/reward_modeling_finetuning.py:9]: Add link and license to the original file. trl is apache 2.0 license

File: python/llm/example/GPU/LLM-Finetuning/RewardModeling/reward_modeling_finetuning.py [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+    model_kwargs = dict(
+        trust_remote_code=True,
+        torch_dtype = torch.bfloat16,
+        use_cache = False,
+    )
+    tokenizer = AutoTokenizer.from_pretrained(model_config.model_name_or_path, use_fast=True)
+    model = AutoModelForSequenceClassification.from_pretrained(
+        model_config.model_name_or_path, num_labels=1, **model_kwargs
+    )
+    model = optimize_model(model, low_bit="fp4")
```
Comment (qiyuangong) [inline python/llm/example/GPU/LLM-Finetuning/RewardModeling/reward_modeling_finetuning.py:49]: Add comments about what we are doing

File: python/llm/example/GPU/LLM-Finetuning/RewardModeling/README.md [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+pip install bitsandbytes==0.43.0
+pip install datasets==2.18.0
+pip install transformers==4.39.1
+pip install trl
+pip install wandb
+```
+
+### 2. Reward Modeling Finetune
+
+Here is how to fine-tune opt-350m on an Intel Max GPU server:
```
Comment (qiyuangong) [inline python/llm/example/GPU/LLM-Finetuning/RewardModeling/README.md:25]: Can we fintune on ARC? if not please specify GPU in Intel GPUs

File: python/llm/example/GPU/LLM-Finetuning/RewardModeling/README.md [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+
+```bash
+# arguments can be reset in the script e.g. model_name_or_path, per_device_train_batch_size and other hyperparameters
```
Comment (qiyuangong) [inline python/llm/example/GPU/LLM-Finetuning/RewardModeling/README.md:28]: Add arguments list, or link

File: python/llm/example/GPU/LLM-Finetuning/RewardModeling/reward_modeling_finetuning.py (hunk 1 above)
Comment (Uxito-Ada) [reply python/llm/example/GPU/LLM-Finetuning/RewardModeling/reward_modeling_finetuning.py:9]: done

**Review by qiyuangong — APPROVED** [review]
LGTM
